src/settings.ts

```diff
@@ -11,7 +11,7 @@ export type ISurveyEnvironment = {
   svgMountContainer: HTMLElement | string,
   stylesSheetsMountContainer: HTMLElement,
 }
-const document = globalThis.document;
+const document = typeof globalThis !== "undefined" ? globalThis.document : self.document;
```

Comment:
Probably we need to change to the `this.document;`

Reply:
Since `this.document` will failed on eslint check due to the possible undefined, I need to do cast as any
```
[6] ERROR in C:\Users\janni\Downloads\survey-library\src\settings.ts
[6] [tsl] ERROR in C:\Users\janni\Downloads\survey-library\src\settings.ts(14,18)
[6]       TS2532: Object is possibly 'undefined'.

Reply:
I meant
```
const document = typeof globalThis !== "undefined" ? globalThis.document : this.document;
```

Reply:
I'm afraid `self` can be undefined under some circumstances